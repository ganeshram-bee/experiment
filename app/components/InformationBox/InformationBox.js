import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import BulbImg from './bulb.svg';
import FlagImg from './ic_redflag.svg';
import Qimg from './ic_question.svg';

const LabelCon = styled.div`
  display: flex;
  position: relative;
  align-items: center;
  border: solid ${props => (props.noBorder ? '0px' : '1px')}
    ${props => (props.strictInfo ? '#f5caae' : '#c8d0ef')};
  background-color: #fff;
  text-align: start;
  border-top-left-radius: 12px;
  border-bottom-right-radius: 12px;
  ${props => (props.btm ? `margin-bottom: ${props.btm}px;` : '')}
  > table {
    width: 100%;
    > tbody > tr > td {
      &:first-child {
        width: 30px;
        text-align: center;
      }
      &:last-child {
        position: relative;
      }
      > img {
        width: 16px;
        position: absolute;
        top: 6px;
        left: 8px;
      }
      > p {
        font-size: 12px;
        color: #6b6f86;
        margin: 0;
        font-weight: 500;
        line-height: 1.67;
        padding: 4px;
        margin-right: 30px;
      }
    }
  }
`;

const InfoP = styled.p`
  font-size: 12px;
  color: #6b6f86;
  margin: 0;
  font-weight: 500;
  line-height: 1.67;
  padding: 4px;
  margin-right: 30px;
  > span {
    font-weight: 600;
    font-size: 13px;
  }
  ${props => (props.strictInfo ? 'font-style:italic;' : null)}
`;

const LiICon = styled.div`
  width: 13px;
  position: absolute;
  right: 3%;
  bottom: 4px;
`;

const HrefLink = styled.span`
  font-size: 12px;
  color: #f2b100;
  margin: 0;
  font-weight: 500;
  line-height: 1.67;
`;

export default function CautionInfo(props) {
  const {
    txt,
    btm,
    strictInfo,
    link,
    hrefTxt,
    onClickTxt,
    onClickHelp,
  } = props;
  return (
    <LabelCon btm={btm} strictInfo={strictInfo}>
      <table>
        <tbody>
          <tr>
            <td>
              <img src={strictInfo ? FlagImg : BulbImg} alt="Info image" />
            </td>
            <td>
              <InfoP strictInfo={strictInfo}>
                <span> 
                    {txt} 
                </span>
              </InfoP>
            </td>
          </tr>
        </tbody>
      </table>
    </LabelCon>
  );
}

CautionInfo.defaultProps = {
  txt: 'Important Instructions',
  strictInfo: false,
};

CautionInfo.propTypes = {
  btm: PropTypes.string,
  txt: PropTypes.string.isRequired,
  strictInfo: PropTypes.bool,
  link: PropTypes.bool,
  hrefTxt: PropTypes.string,
  onClickHelp: PropTypes.func,
  onClickTxt: PropTypes.func,
};
