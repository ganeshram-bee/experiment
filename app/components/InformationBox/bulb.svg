<svg xmlns="http://www.w3.org/2000/svg" width="16" height="20" viewBox="0 0 16 20">
    <defs>
        <style>
            .cls-1,.cls-2{fill:#fff}.cls-1{opacity:0}.cls-3{fill:#ffb400}.cls-4,.cls-5,.cls-6,.cls-7{fill:none}.cls-5{stroke:#484848}.cls-6,.cls-7{stroke-linecap:round;stroke-miterlimit:10;stroke-width:.5px}.cls-6{opacity:.6;stroke:#6b6f86}.cls-7{stroke:#bbbdc4}.cls-10,.cls-8,.cls-9{stroke:none}.cls-9{fill:#484848}.cls-10{fill:#776c50}
        </style>
    </defs>
    <g id="Bulb_for_hint" data-name="Bulb for hint" transform="translate(1051 -129)">
        <g id="bulb">
            <path id="bounding_box" d="M0 0h16v20H0z" class="cls-1" transform="translate(-1051 129)"/>
            <g id="Path_9089" class="cls-2" data-name="Path 9089">
                <path d="M5.942 16.122a2.703 2.703 0 0 1-2.7-2.7v-1.779c0-.66-.533-1.337-1.097-2.053-.266-.338-.541-.688-.734-1.008l-.01-.018-.013-.016C.79 7.718.5 6.866.5 5.942.5 2.942 2.941.5 5.942.5a5.448 5.448 0 0 1 5.443 5.442c0 1.197-.411 2.213-1.293 3.196l-.013.015-.013.017c-.105.139-.218.281-.326.42-.565.716-1.098 1.392-1.098 2.053v1.78c0 1.488-1.211 2.7-2.7 2.7z" class="cls-8" transform="translate(-1049 132)"/>
                <path d="M5.942 15.622c1.213 0 2.2-.987 2.2-2.2v-1.779c0-.834.585-1.577 1.205-2.363.107-.136.218-.277.32-.412l.026-.033.027-.03c.805-.899 1.165-1.782 1.165-2.863A4.948 4.948 0 0 0 5.942 1 4.947 4.947 0 0 0 1 5.942c0 .816.26 1.572.794 2.313l.024.034.021.035c.177.294.442.63.699.956.62.786 1.204 1.529 1.204 2.363v1.78c0 1.212.987 2.2 2.2 2.2m0 1a3.2 3.2 0 0 1-3.2-3.2v-1.78c0-.757-1.172-1.827-1.76-2.803C.377 8 0 7.052 0 5.942a5.942 5.942 0 0 1 11.885 0c0 1.45-.548 2.556-1.42 3.53-.596.785-1.323 1.576-1.323 2.171v1.78a3.2 3.2 0 0 1-3.2 3.2z" class="cls-9" transform="translate(-1049 132)"/>
            </g>
            <path id="Path_8964" d="M3 0a3 3 0 0 1 3 3 3.356 3.356 0 0 1-3 3.387A3.356 3.356 0 0 1 0 3a3 3 0 0 1 3-3z" class="cls-3" data-name="Path 8964" transform="translate(-1046.002 134.998)"/>
            <g id="Path_9090" class="cls-4" data-name="Path 9090">
                <path d="M1 0a1 1 0 0 1 1 1v6H0V1a1 1 0 0 1 1-1z" class="cls-8" transform="translate(-1044 138)"/>
                <path d="M1 .5a.5.5 0 0 0-.5.5v5.5h1V1A.5.5 0 0 0 1 .5M1 0a1 1 0 0 1 1 1v6H0V1a1 1 0 0 1 1-1z" class="cls-10" transform="translate(-1044 138)"/>
            </g>
            <path id="Path_9091" d="M0 0h5" class="cls-5" data-name="Path 9091" transform="translate(-1045.5 144.5)"/>
            <path id="Path_8958" d="M100 11a7.413 7.413 0 0 1 3.867 6.514" class="cls-6" data-name="Path 8958" transform="translate(-1139.481 120.12)"/>
            <path id="Path_8959" d="M68.6 3a7.5 7.5 0 0 1 1.821.226" class="cls-6" data-name="Path 8959" transform="translate(-1111.632 127.216)"/>
            <path id="Path_8960" d="M44.6 4.762a6.585 6.585 0 0 1 1.221-.362" class="cls-6" data-name="Path 8960" transform="translate(-1090.346 125.974)"/>
            <path id="Path_8961" d="M3.373 37.859A7.628 7.628 0 0 1 3 35.518 7.4 7.4 0 0 1 4.312 31.3" class="cls-6" data-name="Path 8961" transform="translate(-1053.45 102.116)"/>
        </g>
    </g>
</svg>
