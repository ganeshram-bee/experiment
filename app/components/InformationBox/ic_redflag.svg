<svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 20 20">
    <defs>
        <style>
            .cls-1{fill:#fff;opacity:0}.cls-2{fill:#f87b93}.cls-3{fill:#eb5874}.cls-4{fill:#6d6daa}
        </style>
    </defs>
    <g id="ic_redflag" transform="translate(-2155.744 -640.24)">
        <path id="Path_9193" d="M0 0h20v20H0z" class="cls-1" data-name="Path 9193" transform="translate(2155.744 640.24)"/>
        <path id="Path_9192" d="M2157.081 644.7c-.187.176 2.944 8.714 2.944 8.714s1.733-1.882 2.89-2.05 7.543.719 8.6.373 3.473-2.454 3.349-2.789-3.026-8.213-3.161-8.265-1.922 2.247-3.033 2.321-7.383-.3-8.294 0a20.735 20.735 0 0 0-3.295 1.696z" class="cls-2" data-name="Path 9192" transform="translate(-.41 -.137)"/>
        <path id="Path_9191" d="M1.995 14.354l-2.147-7.68A7.417 7.417 0 0 1 1.263 5.61l2.575 7.507A12.675 12.675 0 0 0 2 14.353zm10.037-1.584c-.637-.023-1.4-.077-2.2-.133l-.155-.011a70.982 70.982 0 0 0-2.407-.14L4.653 4.863c.6.016 1.265.056 1.91.1h.021c.84.051 1.709.1 2.482.1h.322l2.644 7.708zm2.066-.354l-2.651-7.73a6.383 6.383 0 0 0 2.414-1.736l2.83 7.4-.006.007a19.3 19.3 0 0 1-2.585 2.058z" class="cls-3" data-name="Path 9191" transform="translate(2157.406 638.886)"/>
        <path id="Path_9188" d="M18.723-16.92q.066-.066.022-.221l-2.692-7.6a6.417 6.417 0 0 1-1.368 1.193 6.254 6.254 0 0 1-1.357.674 5.963 5.963 0 0 1-1.5.265 11.333 11.333 0 0 1-1.379.022q-.508-.033-1.523-.122-1.236-.11-2.328-.144a6.04 6.04 0 0 0-2.251.387 5.994 5.994 0 0 0-2.152 1.392l2.56 7.4A6.815 6.815 0 0 1 6.509-14.8a5.832 5.832 0 0 1 1.975-.464 16.446 16.446 0 0 1 1.71 0q.717.044 1.843.2.794.11 1.258.155t1.159.066a5.033 5.033 0 0 0 1.17-.077 8.083 8.083 0 0 0 1.059-.309 3.389 3.389 0 0 0 1.081-.641 6.7 6.7 0 0 0 .959-1.05zM17-25.075l2.692 7.646a1.237 1.237 0 0 1-.177 1.149 7.03 7.03 0 0 1-1.456 1.4 5.566 5.566 0 0 1-1.456.762 6.2 6.2 0 0 1-1.6.265 10.569 10.569 0 0 1-1.479-.011q-.563-.055-1.622-.188-1.125-.155-1.821-.2a12.208 12.208 0 0 0-1.622.022 4.824 4.824 0 0 0-1.81.464 5.544 5.544 0 0 0-1.545 1.149l2.122 6.194a.432.432 0 0 1-.309.619l-.154.044a.464.464 0 0 1-.485-.354l-5.12-14.917v-.044l-.375-1.061a.432.432 0 0 1 .309-.619.42.42 0 0 1 .386.011.509.509 0 0 1 .254.3l.11.331a6.994 6.994 0 0 1 2.416-1.392 6.636 6.636 0 0 1 2.442-.354q1.1.066 2.328.166t2.284.122a5.136 5.136 0 0 0 2.1-.409 5.122 5.122 0 0 0 1.9-1.425.974.974 0 0 1 .949-.331 1 1 0 0 1 .739.661z" class="cls-4" data-name="Path 9188" transform="translate(2155 666)"/>
    </g>
</svg>
