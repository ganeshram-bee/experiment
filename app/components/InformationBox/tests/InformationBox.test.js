import React from 'react';
import { render } from 'react-testing-library';
import { generateImage } from 'jsdom-screenshot';

import InformationBox from '../InformationBox';

describe('<InformationBox />', () => {
  
  // it('has no visual regressions', async () => {
  //   render(<InformationBox />);

  //   const screenshot = await generateImage();
  //   expect(screenshot).toMatchImageSnapshot();
  // });

  it('should match the snapshot', () => {
  	const { container } = render(<InformationBox />);
    expect(container.firstChild).toMatchSnapshot();
  });

  // it('should match the snapshot', () => {
  //   const renderedComponent = renderer.create(<CautionInfo />).toJSON();
  //   expect(renderedComponent).toMatchSnapshot();
  // });

});
