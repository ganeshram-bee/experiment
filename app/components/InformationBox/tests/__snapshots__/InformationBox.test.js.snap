// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InformationBox /> should match the snapshot 1`] = `
<div
  class="InformationBox__LabelCon-sc-1818564-0 fIaWGx"
  data-reactroot=""
>
  <table>
    <tbody>
      <tr>
        <td>
          <img
            alt="Info image"
            src="IMAGE_MOCK"
          />
        </td>
        <td>
          <p
            class="InformationBox__InfoP-sc-1818564-1 dwSvgw"
          >
            <span>
              Important Instructions
            </span>
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
